atomic_reactor/utils/resource_manager.py

```diff
@@ -0,0 +1,187 @@
+"""
+Copyright (c) 2022 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+
+import os
+import logging
+import json
+import paramiko
+from datetime import datetime
+from shlex import quote
+
+ROOT_PATH = 'resource_manager/'
+logger = logging.getLogger(__name__)
+
+
+class Host:
+    '''
+    Remote host abstract class
+    '''
+    def __init__(self, username: str, hostname: str, arch: str):
+        self.username = username
+        self.hostname = hostname
+        self.arch = arch
+        self.slots = []
+        self.free_slots = 0
```

Comment:
Why do we want to count number of free slots. Is not better to just create list or set with indexes to slots which are currently available. Their total count can be then retrieved with len(free_slots). e.g. free_slots=[1,3,4] slot 2 is not available. this is just first idea which came to my mind. Probably can be do in a better way.

Reply:
+1

free_slots is something that's changing often, we probably shouldn't save this value at all and it should be computed every time we need to use hosts.
Probably the same for slots, why not to just return them from method?

Reply:
I don't see anyhting related to slots and free_slots in changes

Reply:
@MartinBasti 
https://github.com/tveverka-RH/atomic-reactor/blob/resource_mngr/atomic_reactor/utils/resource_manager.py#L58
https://github.com/tveverka-RH/atomic-reactor/blob/resource_mngr/atomic_reactor/utils/resource_manager.py#L149